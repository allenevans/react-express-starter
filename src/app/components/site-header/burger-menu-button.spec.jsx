import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import BurgerMenuButton from './burger-menu-button';

const renderer = new ReactShallowRenderer();

describe('burger-menu-button component', () => {
  it('should render the component', () => {
    const tree = renderer.render(<BurgerMenuButton />);

    expect(tree).toMatchSnapshot();
  });

  it('should toggle the state.open property when clicking on the button', () => {
    const wrapper = shallow(<BurgerMenuButton />);
    const controlLabel = wrapper.find('label').first();

    controlLabel.simulate('click', { preventDefault() {} });
    expect(wrapper.state().open).toEqual(true);

    controlLabel.simulate('click', { preventDefault() {} });
    expect(wrapper.state().open).toEqual(false);
  });
});
