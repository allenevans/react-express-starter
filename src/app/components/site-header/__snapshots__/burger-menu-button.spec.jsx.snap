// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`burger-menu-button component should render the component 1`] = `
<React.Fragment>
  <label
    className="burger-menu-button__label"
    htmlFor="burger-menu-button__toggle"
    onClick={[Function]}
  />
  <input
    checked={false}
    className="burger-menu-button__toggle"
    id="burger-menu-button__toggle"
    type="checkbox"
  />
</React.Fragment>
`;
